test/Test.java

```diff
@@ -110,21 +124,21 @@ public static String run(String path, Program p){
         try {
             p.assemble(path);
             if(errorlines != null){
-                return "Expected asssembly error, but successfully assembled " + path;
+                return "Expected assembly error, but successfully assembled " + path;
             }
             p.setup(null,stdin);
             Simulator.Reason r = p.simulate();
-            if(r != Simulator.Reason.NORMAL_TERMINATION){
-                return "Ended abnormally while executing " + path;
+            if(r != Simulator.Reason.NORMAL_TERMINATION && r != Simulator.Reason.CLIFF_TERMINATION){
```

Comment:
Should cliff termination be considered the same as a normal termination for testing purposes?
Should there be a tag (#fall-off / #bottomed-out ?) like #exit to catch it?

Reply:
I used my existing test-suite, and some simple programs assumed a `main`-like behavior where there was no explicit exit.
But I think you are right ant that for a testing purpose we should expect an exit unless stated otherwise.